blockly/src/entities/VariableHUD.java

```diff
@@ -295,9 +301,7 @@ public void clearVariables() {
       Label label = (Label) cell.getActor();
       label.setText("");
     }
-    if (this.monsterTable != null) {
-      monsterTable.remove();
-    }
+    monsterTable.remove();
```

Comment:
@Flamtky du hast bei updateMonsterTable() noch das remove aus der stage/den actors `this.stage.getActors().removeValue(monsterTable, true);`. muss/sollte das hier auch so passieren?

Reply:
Bei clear will ich das nicht aus der Stage entfernen. Weil beim Update möchte ich den Table ersetzen mit einen neuen, aber hier bei clear nur "aufräumen". Ich meine am besten wäre es wenn mab sowieso nicht die Tabelle ersetzt beim Updaten sondern nur die Daten ergänzt. Ich meine wenn du willst könnte ich das noch morgen umändern auf diese Logik, wenn das überhaupt geht.

Reply:
ah. 

hmmm, nee, lass mal. ich habe das gefühl, an dieser ecke ist ohnehin noch einige arbeit nötig. aber wir brauchen das nicht fürs projekt, also würde ich da erstmal keine energie reinstecken.

was man aber nochmal überlegen sollte: stört diese visualisierung bei dem, was wir vorhaben? brauchen wir einen schalter? oder quasi verschiedene starter?

Reply:
> was man aber nochmal überlegen sollte: stört diese visualisierung bei dem, was wir vorhaben? brauchen wir einen schalter? oder quasi verschiedene starter?

ich fände einen eine key (zum beispiel "p") für das ein- und ausblenden besser - und ein paar dummy values, um anzudeuten, dass innerhalb der statischen panels eigentlich noch inhalte stehen sollten.

Reply:
> > was man aber nochmal überlegen sollte: stört diese visualisierung bei dem, was wir vorhaben? brauchen wir einen schalter? oder quasi verschiedene starter?
> 
> ich fände einen eine key (zum beispiel "p") für das ein- und ausblenden besser - und ein paar dummy values, um anzudeuten, dass innerhalb der statischen panels eigentlich noch inhalte stehen sollten.

auch ne idee.

aber ich denke vor allem an unser aktuelles projekt. dort brauchen wir imho diese panels gar nicht. also überhaupt nicht. warum soll da code mitlaufen, den wir nicht brauchen? deshalb der toggle, mit dem man die funktionalität "visualisierung" ein- und abschalten kann.

Reply:
Refactoring: Aktualisieren der Daten statt kompletter Austausch der Tabellen: #1698 
Toggle zum An-/Abschalten der Visualisierung: #1697